### DISCUSSION TITLE
What BLE device

### DISCUSSION THREAD

**Discussion by hooonza** [TheMicDiet/chihiros-led-control#33] [General]
Pls, what BLE device can be used with Home Assistant Green (USB)? Thank you 
Well, the answer is here :-) 
https://www.home-assistant.io/integrations/bluetooth/#known-working-high-performance-adapters

**Comment by TheMicDiet** [comment]
I only tested it on Raspberry Pi 3B which has integrated Bluetooth.

**Reply by hooonza** [reply]
It works with Avantree DG40S (and Home Assistant Green). 
Isn't there a bug in the code? Both red and white switches control the red light, not the white. They all light up white together.

**Reply by TheMicDiet** [reply]
Then your device is probably not recognized and therefore the fallback is used which doesn't work for WRGB. What device are you trying to connect?

**Reply by hooonza** [reply]
It is the Chihiros Aquatic Studio, Universal LED WRGB 800.

**Reply by TheMicDiet** [reply]
Yeah, I think there is currently no support for universal WRGB. Could you provide me the full device name from the chihiros app in device settings?

**Reply by hooonza** [reply]
![Screenshot_20240709-140956](https://github.com/TheMicDiet/chihiros-led-control/assets/154825920/ab95942e-2a2f-4d77-95f7-704d62250bcd)
Well this is a device settings page in the app, thank you.

**Reply by TheMicDiet** [reply]
Hi @hooonza, I just released a new version 0.4.0 which should support the Universal WRGB.

**Reply by hooonza** [reply]
Wow, with the new version it works great, thank you very much!

**Comment by senna1992** [comment]
i can discover all of my chihiros lights (4 different types) using the cm4's (rpi4) inside of my homeassistant yellow.

those being:
-> 1. wrgb universal 800 @ FW 21
-> 2. wrgb2 @ FW 26
-> 3. c2 (white only) @ FW 20
-> 4. backlight panel @ FW 21

the universal wrgb is detected corretly, all others are using "fallback" mode.

as of now, i did not manage to control my lights
will be happy to assist you with data,logs,tests (if wanted)



Conclusion: Rpi4 BT Module is not working, TPlink BT-mini-Adapter does work for: 1 + 4
